### PR TITLE
use docker base image in CI

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- consume prebuilt base image in test.yml rather than building every time
- split test.yml build + test into separate steps (so the location of errors is clearer)
- incidental: add clang-format to base image
## Follow-up work
- scheduled rebuilds of base image
## Sample run
https://github.com/viamrobotics/viam-cpp-sdk/actions/runs/11903253787

### CODE REVIEW

**Comment by lia-viam** [comment]
for my benefit and posterity can you elaborate/eli5 a bit on consuming the pre-built base image? like where is that image coming from, is it manually uploaded rn or could be scheduled in the future, etc

**Comment by abe-winter** [comment]
> where is that image coming from, is it manually uploaded rn or could be scheduled in the future, etc

the image is created by this job https://github.com/viamrobotics/viam-cpp-sdk/actions/workflows/docker.yml

and it gets uploaded to here https://github.com/viamrobotics/viam-cpp-sdk/pkgs/container/cpp-base/307921886?tag=bullseye-amd64

(if you scroll down on that page, note it embeds the git sha so we have a fairly good idea of where this came from; not sure if the sha is attested by the runner, but they're giving us a blue check on the right side of the page; I think some attestation is happening)

could easily add a cron trigger to docker.yml if we wanted

**Comment by abe-winter** [comment]
these docker images are also listed on the side of the repo main page if you need to find them in a hurry

image

**Review by lia-viam — APPROVED** [review]
